### PR TITLE
Daffodil 2230 cleanups

### PR DESCRIPTION
**Pull request by mbeckerle**
As part of work on DAFFODIL-2192 I ran into a bunch of things that are minor cleanups.

I want to get these merged separately so they're not in the review mix for the DAFFODIL-2192 changes. 

DAFFODIL-2230

### CODE REVIEW

File: daffodil-core/src/main/scala/org/apache/daffodil/dsom/AnnotatedSchemaComponent.scala [hunk 1]
```diff
@@ -65,10 +65,21 @@ object ResolvesProperties {
...
+// It is very easy to make the mistake of looking up properties on something like
+// a global def/decl, and then you aren't getting the combining of that with
+// the referencing object and its default format (from its containing schema document).
+//
+// We need to refactor and provide ResolvesScopedProperties, and ResolvesLocalProperties
+// mixins so that non-terms can use different method names (but similar idioms) to get local
+// properties. That way lookup of properties that is supposed to be scoped can't be accidently
+// called on non-terms. We really do not want OO-style polymorphism on property finding
+// across terms and non-terms. Only for the terms. Non-terms want their own local-only
+// method names.
```
Comment (stevedlawrence) [inline daffodil-core/src/main/scala/org/apache/daffodil/dsom/AnnotatedSchemaComponent.scala]: Should we create a ticket for this?

File: eclipse-projects/cli-test/.classpath [hunk 2]
```diff
@@ -53,75 +53,75 @@
...
     <attributes>
-      <attribute name="javadoc_location" value="jar:file:lib_managed/docs/com.ibm.icu/icu4j/icu4j-62.1-javadoc.jar!/"/>
+      <attribute name="javadoc_location" value="jar:file:lib_managed/docs/xerces/xercesImpl/xercesImpl-2.12.0-javadoc.jar!/"/>
     </attributes>
   </classpathentry>
-  <classpathentry sourcepath="lib_managed/srcs/xml-resolver/xml-resolver/xml-resolver-1.2-sources.jar" exported="true" kind="lib" path="lib_managed/jars/xml-resolver/xml-resolver/xml-resolver-1.2.jar"> </classpathentry>
-  <classpathentry sourcepath="lib_managed/srcs/xml-apis/xml-apis/xml-apis-1.4.01-sources.jar" exported="true" kind="lib" path="lib_managed/jars/xml-apis/xml-apis/xml-apis-1.4.01.jar">
+  <classpathentry sourcepath="lib_managed/srcs/org.jdom/jdom2/jdom2-2.0.6-sources.jar" exported="true" kind="lib" path="lib_managed/jars/org.jdom/jdom2/jdom2-2.0.6.jar">
     <attributes>
-      <attribute name="javadoc_location" value="jar:file:lib_managed/docs/xml-apis/xml-apis/xml-apis-1.4.01-javadoc.jar!/"/>
```
Comment (stevedlawrence) [inline eclipse-projects/cli-test/.classpath]: I really wish we could get [DAFFODIL-1983](https://issues.apache.org/jira/browse/DAFFODIL-1983) resolved, seems like these classpath files are constantly out of date.

File: daffodil-core/src/main/scala/org/apache/daffodil/dsom/AnnotatedSchemaComponent.scala (hunk 1 above)
Comment (mbeckerle) [reply daffodil-core/src/main/scala/org/apache/daffodil/dsom/AnnotatedSchemaComponent.scala]: There are quite a lot of cleanups that are listed as TODO in the code. DAFFODIL-1536 is the ticket for cleaning up all the TODOs and FIXMEs in the code. 

This one is an improvement to maintainability and would make the classes enforce the usage patterns better, but I'm disinclined to mess with this until DSOM stabilizes (the backpointers thing), or fix it as part of that.

File: eclipse-projects/cli-test/.classpath (hunk 2 above)
Comment (mbeckerle) [reply eclipse-projects/cli-test/.classpath]: I updated that ticket... we should consider whether eclipseProjects dir should be in the repo at all or if there is some other way to deal with it.

**Pull request merged**